Framework/Crystal/inc/MantidCrystal/FilterPeaks.h

```diff
@@ -44,8 +45,45 @@ class DLLExport FilterPeaks : public API::Algorithm {
   const std::string category() const override;
 
 private:
+  /// Typedef for the function to get the variable we're filtering against
+  typedef std::function<double(const Geometry::IPeak &)> FilterFunction;
+
+  /// Override for algorithm init
   void init() override;
+  /// Override for algorithm exec
   void exec() override;
+
+  /// Get a function to filter peaks by
+  FilterFunction
+  getFilterVariableFunction(const std::string &filterVariable) const;
+
+  /** Filter the input peaks workspace using a comparator and a value selection
+   * function
+   *
+   * The template parameter is a comparator function such as std::less to use to
+   * compare the filterValue against the corresponding value of the peak.
+   *
+   * @param inputWS :: the input peaks workspace containing peaks to be filtered
+   * @param filteredWS :: the output peaks workspace which will contain a subset
+   * of the inputWS
+   * @param filterFunction :: function extracting the value to filter on from
+   * the peak object
+   * @param filterValue :: the value to compare the filter variable against
+   */
+  template <typename Comparator>
+  void filterPeaks(Mantid::API::IPeaksWorkspace_const_sptr inputWS,
```

Comment:
Can we avoid pass the `shared_ptr`s by value as it has to adjust the reference count? We should favour dereferencing the pointers when the function is called and pass in references to the objects directly.